src/core/TimePos.cpp

```diff
@@ -72,7 +72,12 @@ TimePos TimePos::quantize(float bars) const
 	//Offset from the lower position
 	int offset = m_ticks % interval;
 	//1 if we should snap up, 0 if we shouldn't
-	int snapUp = offset / (interval / 2);
+	// Ternary expression is making sure that the snap happens in the direction to
+	// the right even if m_ticks is negative and the offset is exactly half-way
+	// More details on issue #5840 and PR #5847
+	int snapUp = (offset == -(interval / 2))
```

Comment:
I wonder if this could be condensed by using `std::remainder` to get the offset instead of `%`? It feels like there must be a way to do this w/o such a specific condition, but this works if not.

Reply:
What would be the advantage of using `std::remainder`? Wouldn't the currently used modulus operator be easier and faster since we are working with integers?

Reply:
I believe it behaves differently from `%` in what kind of remainder it gives, since there's different ways to do modulo. I was thinking maybe it would somehow let us avoid the ternary. Let's go with the ternary instead though, it's fine that way too.